pkg/state/impl/etcd/etcd.go

```diff
@@ -0,0 +1,571 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package etcd provides an implementation of state.State in etcd.
+package etcd
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/cosi-project/runtime/pkg/state/impl/store"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// State implements state.CoreState.
+type State struct {
+	cli       *clientv3.Client
+	marshaler store.Marshaler
+}
+
+// NewState creates new State with default options.
+func NewState(cli *clientv3.Client) *State {
```

Comment:
do we really need full `*Client`, can we instead use a subset of the interfaces,

e.g.

```go
type EtcdClient interface{
   client.KV
   client.Watcher
}
```

Reply:
done